commit 65f9a3c0e60528a8f63928bee8ddd0bd699c90cc
Author: Koen De Vleeschauwer <kdv@kdvelectronics.eu>
Date:   Sun Jun 1 08:54:34 2025 +0200

    blackmagic

diff --git a/tools/Kconfig b/tools/Kconfig
index e951e25..8070126 100644
--- a/tools/Kconfig
+++ b/tools/Kconfig
@@ -1,5 +1,6 @@
 menu "tools packages"
 
+source "$PKGS_DIR/packages/tools/blackmagic/Kconfig"
 source "$PKGS_DIR/packages/tools/CmBacktrace/Kconfig"
 source "$PKGS_DIR/packages/tools/EasyFlash/Kconfig"
 source "$PKGS_DIR/packages/tools/EasyLogger/Kconfig"
diff --git a/tools/blackmagic/Kconfig b/tools/blackmagic/Kconfig
new file mode 100644
index 0000000..c83e108
--- /dev/null
+++ b/tools/blackmagic/Kconfig
@@ -0,0 +1,65 @@
+# Kconfig file for package blackmagic
+menuconfig PKG_USING_BLACKMAGIC
+    bool "black magic probe: firmware download tool"
+    default n
+
+if PKG_USING_BLACKMAGIC
+
+    config PKG_BLACKMAGIC_PATH
+        string
+        default "/packages/tools/blackmagic"
+
+    config BLACKMAGIC_ENABLE_GPIO
+        bool "gpio drivers"
+        default y
+        help
+            use standard gpio for jtag and swd
+
+    config BLACKMAGIC_ENABLE_RTT
+        bool "rtt support"
+        default y
+        help
+            rtt (real-time terminal) allows terminal input/output between target and host
+
+    config BLACKMAGIC_ENABLE_RTT_IDENT
+        bool "rtt ident"
+        depends on BLACKMAGIC_ENABLE_RTT
+        default n
+        help
+            set optional rtt secret.
+
+    config BLACKMAGIC_RTT_IDENT
+        string "rtt ident"
+        depends on BLACKMAGIC_ENABLE_RTT_IDENT
+        default ""
+        help
+            optional rtt secret. '%' is converted to ' ' space.
+
+    config BLACKMAGIC_ENABLE_DEBUG
+        bool "debug output"
+        default n
+        help
+            If enabled, debug output is written to the aux port.
+            Useful when debugging blackmagic probe.
+
+    choice
+        prompt "Version"
+        help
+            Select the package version
+
+        config PKG_USING_BLACKMAGIC_LATEST_VERSION
+            bool "latest"
+    endchoice
+
+    config PKG_BLACKMAGIC_VER
+       string
+       default "v1.10.2"   if PKG_USING_BLACKMAGIC_V1_10_2
+       default "latest"    if PKG_USING_BLACKMAGIC_LATEST_VERSION
+
+    config BLACKMAGIC_VERSION
+        string
+        default PKG_BLACKMAGIC_VER
+        help
+            hidden string, follows PKG_BLACKMAGIC_VER, used in version.h
+
+endif
diff --git a/tools/blackmagic/package.json b/tools/blackmagic/package.json
new file mode 100644
index 0000000..eb9d2b1
--- /dev/null
+++ b/tools/blackmagic/package.json
@@ -0,0 +1,27 @@
+{
+  "name": "blackmagic",
+  "description": "firmware download tool",
+  "description_zh": "下载器",
+  "enable": "PKG_USING_BLACKMAGIC",
+  "keywords": [
+    "blackmagic"
+  ],
+  "category": "tools",
+  "author": {
+    "name": "koendv",
+    "email": "koen@kdvelectronics.eu",
+    "github": "koendv"
+  },
+  "license": "MIT",
+  "repository": "https://github.com/koendv/blackmagic-rtthread",
+  "homepage": "https://github.com/koendv/blackmagic-rtthread#README.md",
+  "doc": "https://black-magic.org/",
+  "site": [
+    {
+      "version": "latest",
+      "URL": "https://github.com/koendv/blackmagic-rtthread.git",
+      "filename": "",
+      "VER_SHA": "main"
+    }
+  ]
+}
