diff --git a/components/drivers/cputime/Kconfig b/components/drivers/cputime/Kconfig
index 97c2c46259..b2ac749f75 100644
--- a/components/drivers/cputime/Kconfig
+++ b/components/drivers/cputime/Kconfig
@@ -21,7 +21,6 @@ if RT_USING_CPUTIME
         bool "Support Cortex-M CPU"
         default y
         depends on ARCH_ARM_CORTEX_M0 || ARCH_ARM_CORTEX_M3 || ARCH_ARM_CORTEX_M4 || ARCH_ARM_CORTEX_M7
-        select PKG_USING_PERF_COUNTER
     config RT_USING_CPUTIME_RISCV
         bool "Use rdtime instructions for CPU time"
         default y
diff --git a/components/drivers/cputime/cputime_cortexm.c b/components/drivers/cputime/cputime_cortexm.c
index 100910a9f9..d236ae86a3 100644
--- a/components/drivers/cputime/cputime_cortexm.c
+++ b/components/drivers/cputime/cputime_cortexm.c
@@ -23,7 +23,7 @@ static uint64_t cortexm_cputime_getres(void)
 {
     uint64_t ret = 1000UL * 1000 * 1000;
 
-    ret = (ret * (1000UL * 1000)) / SystemCoreClock;
+    ret = (ret * (1000UL * 1000)) / system_core_clock;
     return ret;
 }
 
