diff --git a/components/drivers/spi/sfud/src/sfud.c b/components/drivers/spi/sfud/src/sfud.c
index 2f920cf67a..0d26353f35 100644
--- a/components/drivers/spi/sfud/src/sfud.c
+++ b/components/drivers/spi/sfud/src/sfud.c
@@ -841,6 +841,9 @@ static sfud_err reset(const sfud_flash *flash) {
     cmd_data[1] = SFUD_CMD_RESET;
     result = spi->wr(spi, &cmd_data[1], 1, NULL, 0);
 
+    /* allow flash to settle */
+    rt_thread_delay(15);
+
     if (result == SFUD_SUCCESS) {
         result = wait_busy(flash);
     }
